subsys/dfu/img_util/flash_img.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __FLASH_IMG_H__
+#define __FLASH_IMG_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct flash_img_context {
+	u8_t buf[CONFIG_IMG_BLOCK_BUF_SIZE];
+	struct device *dev;
+	u32_t bytes_written;
```

Comment:
This is still a u32_t instead of a size_t -- can you explain why that is better, since the flash API counts bytes in terms of size_t?